### PR TITLE
Backspace support

### PR DESCRIPTION
**Pull request by justinfagnani**

### CODE REVIEW

**Comment by justinfagnani** [comment]
@chuckh created a PR so I can comment

File: elements/designer-stage/designer-stage.html [hunk 1]
```diff
@@ -425,10 +429,19 @@
             return;
         }
 
-        this.frame.client.moveCursor(move)
-          .then(function(response) {
-            this._updateCaret(response.rect);
-          }.bind(this));
+        if (move != 'backspace') {
```
Comment (justinfagnani) [inline elements/designer-stage/designer-stage.html]: let's swap these cases so the default is a move: `if (move === 'backspace') {...`

File: elements/designer-stage/designer-stage.html [hunk 2]
```diff
@@ -425,10 +429,19 @@
...
+          this.frame.client.moveCursor(move)
+            .then(function(response) {
+              this._updateCaret(response.rect);
+            }.bind(this));
+        } else {
+          var keyCode = event.keyCode;
+          var char = String.fromCharCode(keyCode);
```
Comment (justinfagnani) [inline elements/designer-stage/designer-stage.html]: why do you need the `char` here?

File: src/protocol/DocumentServer.js [hunk 3]
```diff
@@ -222,6 +223,33 @@ define('polymer-designer/protocol/DocumentServer', [
       });
     }
 
+    backspaceText(request) {
+      let text = request.message.text;
+      let node = document.createTextNode(text);
```
Comment (justinfagnani) [inline src/protocol/DocumentServer.js]: looks like this `node` is never used

File: elements/designer-stage/designer-stage.html (hunk 1 above)
Comment (chuckh) [inline elements/designer-stage/designer-stage.html]: makes sense.

File: elements/designer-stage/designer-stage.html (hunk 2 above)
Comment (chuckh) [inline elements/designer-stage/designer-stage.html]: We don't. I will change.

File: src/protocol/DocumentServer.js (hunk 3 above)
Comment (chuckh) [inline src/protocol/DocumentServer.js]: I tried it without it and it does not work. Cursor does not move when backspacing.

**Comment by chuckh** [comment]
I made a new commit with Removed char and swapped backspace case. You can see the changes at
 at https://github.com/chuckh/designer/commit/8f603d50d1d1c1f4a815901938f77dfa692e72a4.

File: src/protocol/DocumentServer.js (hunk 3 above)
Comment (justinfagnani) [inline src/protocol/DocumentServer.js]: That is very strange, you are not using this `node` anywhere, you are writing a new value to `node` on line 238. What happens if you rename `node` to `what`? Is the simple act of creating a Text node causing something else to happen?

Comment (justinfagnani) [inline src/protocol/DocumentServer.js]: ping? Have you looked at what's going on w/ `node`?

Comment (chuckh) [inline src/protocol/DocumentServer.js]: @justinfagnani. I did some more testing. If I rename to `node` to `what` it does not work (cursor does not move, just like when removing 'node'.  Also found with `node` it works when you backspace only, but if you add some text then backspace , backspacing stops working.

I will play with it some more. Not having much luck so far. Probably will not have a chance till tomorrow. Converting Road to Polymer to 0.9 currently.

Comment (justinfagnani) [inline src/protocol/DocumentServer.js]: That's so very weird. I definitely want to understand what's going on there, so when you can get this rebased, let me know and I'll debug locally.

And of course, the Road to Polymer is a little more important to the community :D :+1:

**Pull request closed without merging**